[Task] Various Changes

martineiber opened the pull request:
## Changes in this pull request
 - Update http code form 400 to 422 for invalid Argument.
 - Return symfony violation to client.
 - Fix Schema.

## Additional info

martineiber committed:
Update http code form 400 to 422 for invalid Argument.

martineiber committed:
Return symfony violation to client.

martineiber committed:
Fix Schema.

martineiber committed:
Apply php-cs-fixer changes

martineiber committed:
Fix line

martineiber committed:
Apply php-cs-fixer changes

martineiber committed:
Fix Stan

lukmzig commented on src/Asset/OpenApi/Attribute/Parameter/Query/ContainParameter.php:
```diff
@@ -20,13 +20,13 @@
 #[Attribute(Attribute::TARGET_METHOD)]
 final class ContainParameter extends QueryParameter
```
If we are already changing this, can we replace it completely with `Pimcore\Bundle\StudioBackendBundle\OpenApi\Attribute\Parameter\Query\BoolParameter`?

lukmzig commented on src/Asset/OpenApi/Attribute/Parameter/Query/CoverParameter.php:
```diff
@@ -20,13 +20,13 @@
 #[Attribute(Attribute::TARGET_METHOD)]
 final class CoverParameter extends QueryParameter
```
If we are already changing this, can we replace it completely with `Pimcore\Bundle\StudioBackendBundle\OpenApi\Attribute\Parameter\Query\BoolParameter`?

lukmzig commented on src/Asset/OpenApi/Attribute/Parameter/Query/ForceResizeParameter.php:
```diff
@@ -20,13 +20,13 @@
 #[Attribute(Attribute::TARGET_METHOD)]
 final class ForceResizeParameter extends QueryParameter
```
If we are already changing this, can we replace it completely with Pimcore\Bundle\StudioBackendBundle\OpenApi\Attribute\Parameter\Query\BoolParameter?

lukmzig commented on src/Asset/OpenApi/Attribute/Parameter/Query/FrameParameter.php:
```diff
@@ -20,13 +20,13 @@
 #[Attribute(Attribute::TARGET_METHOD)]
 final class FrameParameter extends QueryParameter
```
If we are already changing this, can we replace it completely with Pimcore\Bundle\StudioBackendBundle\OpenApi\Attribute\Parameter\Query\BoolParameter?

lukmzig reviewed (changes requested):
Small suggestion to replace some Asset Query Parameters to more Generic Query BoolParameter if we are already changing them. Maybe also change  `?bool $example = null,` in `Pimcore\Bundle\StudioBackendBundle\OpenApi\Attribute\Parameter\Query\BoolParameter` and make it not nullable

Rest LGTM

martineiber committed:
Replace some Parameter with BoolParameter

martineiber commented:
> Small suggestion to replace some Asset Query Parameters to more Generic Query BoolParameter if we are already changing them. Maybe also change `?bool $example = null,` in `Pimcore\Bundle\StudioBackendBundle\OpenApi\Attribute\Parameter\Query\BoolParameter` and make it not nullable
> 
> Rest LGTM

Good Point! I have removed those Parameter and replaced them with the BoolParameter.

martineiber committed:
Merge branch '1.x' into fix-response
